vanilla/vanilla
# Fix cache busting of dynamically imported modules

## Chunk-hash as a cache buster
Closes https://github.com/vanilla/vanilla/issues/7662

- Reverts the changes from #7661
- Adds the hash of a dynamic chunks contents as a query parameter at the end of its url.

### To verify

- Checkout this branch
- Navigate to a page where the rich editor mounts.
- Inspect the head of the document.
- Notice that the dynamic chunk `plugins/rich-editor/js/webpack/chunks/mountEditor.min.js` has this query parameter on the end of it.

![image](https://user-images.githubusercontent.com/1770056/44542379-a1a7a500-a6da-11e8-83cf-39518eb117e9.png)

## Webpack jsonp function name

Closes https://github.com/vanillaforums/immediate-media/issues/4

A conflict was reported by a client with us both using the same webpackJsonP global name. Webpack stores it's list of modules in a global variable for all chunks to reference, but we were using the default `window.webpackJsonp`. A client was also using the default and having both scripts creates a conflict preventing ours from loading.

The remedy for this is setting the `library` name or the the `jsonpfunction` https://webpack.js.org/configuration/output/#output-jsonpfunction.
